assets/app/views/browse/edit-build-config.html

```diff
@@ -0,0 +1,257 @@
+                            <div>
+                              <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                              <input class="form-control"
+                                id="sourceUrl"
+                                name="sourceUrl"
+                                ng-model="updatedBuildConfig.spec.source.git.uri"
+                                type="text"
+                                aria-describedby="from_source_help">
+                            </div>
+                          </div>
+                          <div>
+                            <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.sourceUrl)">Git repository should be a URL.</span>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div ng-if="updatedBuildConfig.spec.source.type == 'Git'">
+                            <label for="sourceRef">Git Repository Ref</label>
+                            <div>
+                              <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                              <input class="form-control"
+                                id="sourceRef"
+                                name="sourceRef"
+                                type="text"
+                                ng-model="updatedBuildConfig.spec.source.git.ref"
+                                aria-describedby="from_source_help"
+                                placeholder="{{updatedBuildConfig.spec.source.git.ref === undefined ? 'master' : updatedBuildConfig.spec.source.git.ref}}">
+                            </div>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div ng-if="updatedBuildConfig.spec.source.type == 'Git'">
+                            <label for="sourceContextDir">Git Context Dir</label>
+                            <div>
+                              <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                              <input class="form-control"
+                                id="sourceContextDir"
+                                name="sourceContextDir"
+                                type="text"
+                                ng-model="updatedBuildConfig.spec.source.git.contextDir"
+                                aria-describedby="from_source_help"
+                                placeholder="/">
+                            </div>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div class="checkbox" ng-if="buildConfig.spec.strategy.type === 'Docker'">
```

Comment:
Nit: I'd put the `ng-if` on the parent `form-group` div so an empty div isn't added to the page for other strategies.